### PR TITLE
Update API docs

### PR DESCRIPTION
**Pull request by lahirusamith**
## Purpose
$tiitle

Fix for: https://github.com/ballerina-platform/ballerina-standard-library/issues/1463

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: sql-ballerina/Module.md [hunk 1]
```diff
@@ -156,16 +156,20 @@ string|int? generatedKey = ret.lastInsertId;
...
+record(i.e., the `ID` column in the result can be mapped to the `id`property in the record). Additional Column names
+added to the returned record as in the SQL query. If the record is defined as close record, only defined fields in the
+record are returned or gives an error. Next, the `SELECT` query is executed via the `query` remote function of the client.
+Once the query is executed, each data record can be retrieved by looping the result set. The `stream`returned by the
+select operation holds a pointer to the actual data in the database and it loads data from the table only when it is
+accessed. This stream can be iterated only once.
 
 ```ballerina
-// Define a type to represent the results.
+// Define a open record type to represent the results.
```
Comment (niveathika) [inline sql-ballerina/Module.md:172]:
```suggestion
// Define an open record type to represent the results.
```

**Comment by daneshk** [comment]
We need to update this code https://github.com/ballerina-platform/module-ballerina-sql/pull/271/files#diff-bf6e7e22fe07ce6b711e9e35eeed9452f5e945927a929ad7daad813dd0080405R244-R249 to read record

File: sql-ballerina/Module.md [hunk 2]
```diff
@@ -232,30 +260,30 @@ error? e = resultStream.close();
...
-This example demonstrates modifying data by executing an `UPDATE` statement via the `execute` remote function of 
+This sample demonstrates modifying data by executing an `UPDATE` statement via the `execute` remote function of 
 the client.
 
 ```ballerina
 int age = 23;
 sql:ParameterizedQuery query = `UPDATE students SET name = 'John' 
                                 WHERE age = ${age}`;
-sql:ExecutionResult|sql:Error ret = check dbClient->execute(query);
+sql:ExecutionResult|sql:Error result = check dbClient->execute(query);
```
Comment (daneshk) [inline sql-ballerina/Module.md:270]:
```suggestion
sql:ExecutionResult result = check dbClient->execute(query);
```

Comment (daneshk) [reply sql-ballerina/Module.md:270]: no need to error and check both

File: sql-ballerina/Module.md [hunk 3]
```diff
@@ -232,30 +260,30 @@ error? e = resultStream.close();
...
 
-This example demonstrates deleting data by executing a `DELETE` statement via the `execute` remote function of 
+This sample demonstrates deleting data by executing a `DELETE` statement via the `execute` remote function of 
 the client.
 
 ```ballerina
 string name = "John";
 sql:ParameterizedQuery query = `DELETE from students WHERE name = ${name}`;
-sql:ExecutionResult|sql:Error ret = check dbClient->execute(query);
+sql:ExecutionResult|sql:Error result = check dbClient->execute(query);
```
Comment (daneshk) [inline sql-ballerina/Module.md:281]:
```suggestion
sql:ExecutionResult result = check dbClient->execute(query);
```

File: sql-ballerina/Module.md [hunk 4]
```diff
@@ -143,44 +143,71 @@ string name = "Kate";
...
-property in the record). Next, the `SELECT` query is executed via the `query` remote function of the client by passing that 
-result set type. Once the query is executed, each data record can be retrieved by looping the result set. The `stream` 
-returned by the select operation holds a pointer to the actual data in the database and it loads data from the table 
-only when it is accessed. This stream can be iterated only once. 
+This sample demonstrates querying data from a table in a database.
+First, a type is created to represent the returned result set. This record can be defined as an open or a closed record
+according to the requirement. If an open record is defined, the returned stream type will include both defined fields
+in the record and additional database columns fetched by the SQL query which are not defined in the record.
+Note the mapping of the database column to the returned record's property is case-insensitive if it is defined in the
+record(i.e., the `ID` column in the result can be mapped to the `id`property in the record). Additional Column names
```
Comment (daneshk) [inline sql-ballerina/Module.md:164]:
```suggestion
record(i.e., the `ID` column in the result can be mapped to the `id` property in the record). Additional Column names
```

Comment (lahirusamith) [reply sql-ballerina/Module.md]: corrected in https://github.com/ballerina-platform/module-ballerina-sql/pull/271/commits/4b7a5e42e12db6ff25d5d8ff66db0684213fb0b3

File: sql-ballerina/Module.md [hunk 5]
```diff
@@ -143,44 +143,71 @@ string name = "Kate";
...
+added to the returned record as in the SQL query. If the record is defined as a close record, only defined fields in the
+record are returned or gives an error when additional columns present in the SQL query. Next, the `SELECT` query is executed
+via the `query` remote function of the client. Once the query is executed, each data record can be retrieved by looping 
+the result set. The `stream`returned by the select operation holds a pointer to the actual data in the database and it
```
Comment (daneshk) [inline sql-ballerina/Module.md]:
```suggestion
the result set. The `stream` returned by the select operation holds a pointer to the actual data in the database and it 
```

Comment (lahirusamith) [reply sql-ballerina/Module.md]: corrected in https://github.com/ballerina-platform/module-ballerina-sql/pull/271/commits/4b7a5e42e12db6ff25d5d8ff66db0684213fb0b3

File: sql-ballerina/Module.md [hunk 6]
```diff
@@ -189,12 +216,12 @@ error? e = resultStream.forEach(function(Student student) {
 ```
```
Comment (daneshk) [inline sql-ballerina/Module.md:216]: Since the only change is closed record type. others are the same. let's add only one code sample

Comment (lahirusamith) [reply sql-ballerina/Module.md:216]: SQL Query is also changed. that is why I added another code sample.

Comment (daneshk) [reply sql-ballerina/Module.md:216]: yes, let's remove it. the sample doesn't reflect our intention clearly.

**Review by daneshk — APPROVED** [review]

**Pull request merged**